src/mocks/mockReducer.tsx

```diff
@@ -0,0 +1,18 @@
+import { BrowserRouter as Router } from 'react-router-dom'
+import { render } from '@testing-library/react'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+
+const queryClient = new QueryClient()
+
+function reducer(ui: any, { preloadedState, store, ...renderOptions }: any = {}) {
+  function Wrapper({ children }: any) {
+    return (
+      <Router>
+        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+      </Router>
+    )
+  }
+  return render(ui, { wrapper: Wrapper, ...renderOptions })
+}
+
```

Comment:
Curiosità: c'è un motivo per cui è stato scelto il nome `reducer` per questa funzione?

Reply:
By searching i found it called reducer so i leave it like that. Do we need to rename it?

Reply:
`reducer` sounds a bit confusing. I would try something more descriptive like `renderWithApplicationContext` or something like that if you have a better idea